python/cudf/cudf/io/json.py

```diff
@@ -136,18 +285,64 @@ def read_json(
     return df
 
 
-def maybe_return_nullable_pd_obj(cudf_obj):
+def _maybe_return_nullable_pd_obj(
+    cudf_obj: cudf.DataFrame | cudf.Series,
+) -> pd.DataFrame | pd.Series:
     try:
         return cudf_obj.to_pandas(nullable=True)
     except NotImplementedError:
         return cudf_obj.to_pandas(nullable=False)
 
 
+def _dtype_to_names_list(col: ColumnBase) -> list[tuple[abc.Hashable, Any]]:
+    if isinstance(col.dtype, cudf.StructDtype):
+        return [
+            (name, _dtype_to_names_list(child))
+            for name, child in zip(col.dtype.fields, col.children)
+        ]
+    elif isinstance(col.dtype, cudf.ListDtype):
+        return [("", _dtype_to_names_list(child)) for child in col.children]
+    return []
+
+
+@acquire_spill_lock()
+def _plc_write_json(
```

Comment:
Maybe we should add a `_plc_read_json` too?

Reply:
I added this since we needed to call `write_json` twice. For the read, we only called `read_json` and `chunked_read_json` once respectively so I don't think there's a need to make a separate function for it.